dash/dash-renderer/src/utils/clientsideFunctions.ts

```diff
@@ -0,0 +1,29 @@
+import {updateProps, notifyObservers} from '../actions/index';
+import {getPath} from '../actions/paths';
+
+const set_props = (updates: []) => {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const ds = (window.dash_stores = window.dash_stores || []);
+    for (let y = 0; y < ds.length; y++) {
+        const {dispatch, getState} = ds[y];
+        const {paths} = getState();
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        updates.forEach(({id, ...props}) => {
```

Comment:
I think the api for this could be simpler without the batching of updates and extracting the id from the props into a function argument.
The function signature would then look like:
```
set_props(id: string, props: {[k: string]: any}): void;
```

Reply:
Maybe, unless we would want to be able to send batches of updates and not trigger the renderer until the props have been updated?

Reply:
For batching to work it needs a `import {batch} from 'react-redux'`, or in react 18 it would be kinda automatic in effects and lifecycle methods. But it's only for calling within a react context, which might not always be the case here.

Reply:
Yeah, for this, it would be in the context of JS, however, I thought with the function coming from within the React lifecycle, this would still be a react response?

Reply:
Inside a clientside callback there is a react context, but since this is exposed globally it may be called from event handlers or anywhere really.
I think in the most simple cases it would be to update a single component like a toast message and multiple props on the component can still be set at once.
This API will also be implemented on the Python side for updating non output props inside callbacks, a simpler API will be better in that case and to make sure the id is provided.

Reply:
Ok, yeah, I can change it, was just thinking in general how the callbacks work with being able to pass multiple at the same time.